src/Plugin/migrate/source/YAML.php

```diff
@@ -0,0 +1,175 @@
+<?php
+
+namespace Drupal\migrate_default_content\Plugin\migrate\source;
+
+use Drupal\migrate\MigrateException;
+use Drupal\migrate\Plugin\migrate\source\SourcePluginBase;
+use Drupal\migrate\Plugin\MigrationInterface;
+
+/**
+ * Source for YAML.
+ *
+ * If the YAML file contains non-ASCII characters, make sure it includes a
+ * UTF BOM (Byte Order Marker) so they are interpreted correctly.
+ *
+ * @MigrateSource(
+ *   id = "yaml"
```

Comment:
Dont know how to go public with a new module for drupal, can you please give me a hint?